deblur/workflow.py

```diff
@@ -7,20 +7,35 @@
 # -----------------------------------------------------------------------------
 
 
-def trim_seqs(seqs_fp, trim_len):
-    """Step 1: trim FASTA sequences to specified length"""
-    pass
+from os.path import splitext
+from bfillings.vsearch import vsearch_dereplicate_exact_seqs
 
 
-def dereplicate_seqs(seqs_fp):
-    """Step 2a: dereplicate FASTA sequences using VSEARCH"""
+def trim_seqs(seqs_fp, trim_len):
+    """Step 1: trim FASTA sequences to specified length"""
     pass
 
 
-def remove_singletons_seqs(seqs_fp):
-    """Step 2b: remove singletons from dereplicated FASTA
-       file using VSEARCH"""
-    pass
+def dereplicate_seqs(seqs_fp,
+                     output_fp,
+                     min_size=2):
+    """Step 2a: dereplicate FASTA sequences and remove
+       singletons using VSEARCH
+
+    Parameters
+    ----------
+    seqs_fp : string
+        filepath to FASTA sequence file
+    output_fp : string
+        filepath to dereplicated FASTA file
+    """
+    log_name = "%s.log" % splitext(output_fp)[0]
```

Comment:
I think the current structure makes sense, because you're using python's
splitext function, which removes the extension from the file. I would
suggest live as it is.

2015-04-13 11:18 GMT-07:00 Evguenia Kopylova <notifications@github.com>:

> In deblur/workflow.py
> <https://github.com/biocore/deblur/pull/19#discussion_r28263743>:
>
> > -       file using VSEARCH"""
> > -    pass
> > +def dereplicate_seqs(seqs_fp,
> > +                     output_fp,
> > +                     min_size=2):
> > +    """Step 2a: dereplicate FASTA sequences and remove
> > +       singletons using VSEARCH
> > +
> > +    Parameters
> > +    ----------
> > +    seqs_fp : string
> > +        filepath to FASTA sequence file
> > +    output_fp : string
> > +        filepath to dereplicated FASTA file
> > +    """
> > +    log_name = "%s.log" % splitext(output_fp)[0]
>
> the log file will be called XXX.YYY.log
>
> I can change it to add '.log' to the end of the input file name, without
> assuming the last string after '.' is the extension ..
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/biocore/deblur/pull/19/files#r28263743>.
>



-- 
Jose Navas